app/supportAssignment/services/supportSelectors.js

```diff
@@ -0,0 +1,422 @@
+					var supportStaff = supportStaffList.list[supportStaffId];
+
+					if (processedSupportStaffIds.indexOf(supportStaffId) > -1) {
+						return;
+					}
+
+					supportStaff.supportStaffId = supportStaff.id;
+					sectionGroup.readerAssignmentOptions.other.push(supportStaff);
+				});
+
+				// Add instructor preference comment to sectionGroup by following the relationship
+				// sectionGroups -> instructorPreferences -> instructorSupportCallResponses
+				instructorPreferences.ids.forEach(function(preferenceId) {
+					var preference = instructorPreferences.list[preferenceId];
+					if (preference.sectionGroupId == sectionGroup.id) {
+						instructorSupportCallResponses.ids.forEach(function(responseId) {
+							var response = instructorSupportCallResponses.list[responseId];
+							if (response.instructorId == preference.instructorId) {
+								sectionGroup.instructorPreferenceComment = response.generalComments;
+							}
+						});
+					}
+				});
+
+				var generatedSections = self.generateSections(sectionGroup, sections, supportAssignments, supportStaffList);
+				sectionGroup.sections = _array_sortByProperty(generatedSections, ["sequenceNumber"]);
+
+				newSectionGroups.push(sectionGroup);
+			});
+
+			var sortedSectionGroups = _array_sortByProperty(newSectionGroups, ["subjectCode", "courseNumber"]);
+
+			return sortedSectionGroups;
+		},
+		generateSections: function(sectionGroup, sections, supportAssignments, supportStaffList) {
+			var self = this;
+			var generatedSections = [];
+
+			sections.ids.forEach(function(sectionId) {
+				var section = sections.list[sectionId];
+				if (section.sectionGroupId != sectionGroup.id) {
+					return;
+				}
+
+				// Add assignments to section
+				section.readerAssignmentOptions = sectionGroup.readerAssignmentOptions;
+				section.teachingAssistantAssignmentOptions = sectionGroup.teachingAssistantAssignmentOptions;
+
+				if (!supportAssignments.bySectionIds[sectionId]) {
+					supportAssignments.bySectionIds[sectionId] = [];
+				}
+
+				section.supportAssignments = [];
+
+				supportAssignments.bySectionIds[sectionId].forEach( function(assignmentId) {
+					var supportAssignment = supportAssignments.list[assignmentId];
+
+					// Ensure preference is relevant to sectionGroup
+					if (supportAssignment.sectionId != section.id) {
+						return;
+					}
+
+					// Add supportStaff data
+					supportAssignment = self.addSupportStaffData(supportAssignment, supportStaffList);
+
+					if (supportAssignment.appointmentType == "teachingAssistant") {
+						supportAssignment.viewType = "Teaching Assistants";
+					} else if (supportAssignment.appointmentType == "reader") {
+						supportAssignment.viewType = "Readers";
+					}
+
+					section.supportAssignments.push(supportAssignment);
+				});
+
+				generatedSections.push(section);
+			});
+
+			return generatedSections;
+		},
+		generateSupportStaffList: function (
+			supportAssignments,
```

Comment:
Place 'supportAssignments' on same line as 'function', then line up remaining lines with the start of the word 'supportAssignments'